mergify_engine/web/api/queues.py

```diff
@@ -319,3 +360,272 @@ async def repository_queues(
+            redis_cache, repo["owner"]["id"]
+        )
+        installation = context.Installation(
+            installation_json, sub, client, redis_cache, redis_queue
+        )
+        repository_ctxt = installation.get_repository_from_github_data(repo)
+
+        qf = await freeze.QueueFreeze.get(repository_ctxt, queue_name)
+        if qf is None:
+            raise fastapi.HTTPException(
+                status_code=404,
+                detail=f'The queue "{queue_name}" does not exists or is not currently frozen.',
+            )
+
+        return QueueFreezeResponse(
+            queue_freeze=[
+                QueueFreeze(
+                    name=qf.name,
+                    reason=qf.reason,
+                    user_app_name=qf.user_app_name,
+                    freeze_date=qf.freeze_date,
+                )
+            ],
+            description=f'The queue "{queue_name}" is currently frozen.',
+        )
+
+
+@router.get(
+    "/repos/{owner}/{repository}/queues/freezes",  # noqa: FS003
+    summary="Get the list of frozen queues",
+    description="Get the list of frozen queues inside the requested repository",
+    response_model=QueueFreezeResponse,
+)
+async def get_list_queue_freeze(
+    owner: github_types.GitHubLogin = fastapi.Path(  # noqa: B008
+        ..., description="The owner of the repository"
+    ),
+    repository: github_types.GitHubRepositoryName = fastapi.Path(  # noqa: B008
+        ..., description="The name of the repository"
+    ),
+    redis_cache: utils.RedisCache = fastapi.Depends(  # noqa: B008
+        redis.get_redis_cache
+    ),
+    redis_queue: utils.RedisQueue = fastapi.Depends(  # noqa: B008
+        redis.get_redis_queue
+    ),
+    installation_json: github_types.GitHubInstallation = fastapi.Depends(  # noqa: B008
+        security.get_installation
+    ),
+) -> QueueFreezeResponse:
+    async with github.aget_client(installation_json) as client:
+        try:
+            # Check this token has access to this repository
+            repo = typing.cast(
+                github_types.GitHubRepository,
+                await client.item(f"/repos/{owner}/{repository}"),
+            )
+        except (http.HTTPNotFound, http.HTTPForbidden, http.HTTPUnauthorized):
+            raise fastapi.HTTPException(status_code=404)
+
+        sub = await subscription.Subscription.get_subscription(
+            redis_cache, repo["owner"]["id"]
+        )
+        installation = context.Installation(
+            installation_json, sub, client, redis_cache, redis_queue
+        )
+        repository_ctxt = installation.get_repository_from_github_data(repo)
+
+        qf_response = QueueFreezeResponse()
+        qf_response.queue_freeze = [
+            qf async for qf in freeze.QueueFreeze.get_all(repository_ctxt)
+        ]
+        if qf_response.queue_freeze:
+            qf_response.description = (
+                f"there is {len(qf_response.queue_freeze)} queue(s) currently frozen."
```

Comment:
```suggestion
                f"There is {len(qf_response.queue_freeze)} queue(s) currently frozen."
```
I'm sure you can compute if the `s` is needed or not 😉